src/vdaf.rs

```diff
@@ -192,23 +187,98 @@ pub enum PrepareTransition<S, M, O> {
     Fail(VdafError),
 }
 
-/// An aggregate share that can merged with aggregate shares of the same type.
-pub trait Aggregatable {
+/// An aggregate share resulting from aggregating output shares together that
+/// can merged with aggregate shares of the same type.
+pub trait Aggregatable: Clone + Debug + Serialize + DeserializeOwned {
+    /// Type of output shares that can be accumulated into an aggregate share.
+    type OutputShare;
+
+    /// Aggregates a sequence of output shares into a new Aggregatable
+    fn aggregate<M: IntoIterator<Item = Self::OutputShare>>(
+        output_shares: M,
+    ) -> Result<Self, VdafError>;
+
     /// Update an aggregate share by merging it with another (`agg_share`).
     fn merge(&mut self, agg_share: &Self) -> Result<(), VdafError>;
+
+    /// Update an aggregate share by merging it with an output share
```

Comment:
nit (aligns the language with the comment above)
```suggestion
    /// Update an aggregate share by adding `output_share`.
```